pkg/utils/powers_of_two.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+// RoundUpPowerOfTwo returns the next power of two greater than or equal to input.
+func RoundUpPowerOfTwo(input int) int {
+	k := 1
+	for k < input {
+		k = k << 1
+	}
+	return k
+}
+
+// RoundDownPowerOfTwo returns the next power of two less than or equal to input.
+func RoundDownPowerOfTwo(v int) int {
+	c := RoundUpPowerOfTwo(v)
+	if c == v {
+		return c
+	}
+	return c / 2
+}
+
+// RoundUpPowerOfTwoU rounds input up to the next power of two.
+// Examples:
+//
+//	RoundUpPowerOfTwoU(1) = 2
+//	RoundUpPowerOfTwoU(2) = 4
+//	RoundUpPowerOfTwoU(5) = 8
+func RoundUpPowerOfTwoU(input uint64) uint64 {
```

Comment:
[Non Blocking][Question]
Whats the difference between this and:

```go
func RoundUpPowerOfTwo(input int) int {
	k := 1
	for k < input {
		k = k << 1
	}
	return k
}
```

aside from the sign?

Reply:
good question. Unsigned vs signed is the rationale for two separate function definitions. As for why the implementations are different 🤷‍♂️ 

The unsigned implementation isn't particularly readable to me so I assume that it intentionally traded-off readability for performance.

Ref:
1. signed: https://www.tutorialspoint.com/golang-program-to-round-up-the-next-highest-power-of-2
2. unsigned (approach 4): https://www.techiedelight.com/round-next-highest-power-2/

I'm inclined to update the less readable unsigned implementation to the signed implementation but would need to run benchmarks to make sure performance isn't significantly different.

Reply:
agreed, if it's not giving us much performance, probably better sacrifice it for readability :+1: 
@evan-forbes When you have time, please let us know what you think :cat:

Reply:
yeah, we can rewrite it, I don't think it was about performance, and even it is its probably not that much